model/task/db.go

```diff
@@ -1365,31 +1453,85 @@ func FindAllFirstExecution(q db.Q) ([]Task, error) {
 }
 
 // Find returns all tasks that satisfy the query it also filters out display tasks from the results.
-func Find(query db.Q) ([]Task, error) {
+func Find(filter bson.M) ([]Task, error) {
 	tasks := []Task{}
+	_, exists := filter[DisplayOnlyKey]
+	if !exists {
+		filter[DisplayOnlyKey] = bson.M{"$ne": true}
+	}
+	query := db.Query(filter)
 	err := db.FindAllQ(Collection, query, &tasks)
 	if adb.ResultsNotFound(err) {
 		return nil, nil
 	}
 
-	filtered := []Task{}
+	return tasks, nil
+}
 
-	// remove display tasks from results
-	for idx := range tasks {
-		t := tasks[idx]
-		if t.DisplayOnly {
-			continue
-		}
-		filtered = append(filtered, t)
+func FindWithFields(filter bson.M, fields ...string) ([]Task, error) {
+	tasks := []Task{}
+	_, exists := filter[DisplayOnlyKey]
+	if !exists {
+		filter[DisplayOnlyKey] = bson.M{"$ne": true}
+	}
+	query := db.Query(filter).WithFields(fields...)
+	err := db.FindAllQ(Collection, query, &tasks)
+	if adb.ResultsNotFound(err) {
+		return nil, nil
+	}
 
+	return tasks, nil
+}
+
+func FindWithSort(filter bson.M, sort []string) ([]Task, error) {
+	tasks := []Task{}
+	_, exists := filter[DisplayOnlyKey]
+	if !exists {
+		filter[DisplayOnlyKey] = bson.M{"$ne": true}
+	}
+	query := db.Query(filter).Sort(sort)
+	err := db.FindAllQ(Collection, query, &tasks)
+	if adb.ResultsNotFound(err) {
+		return nil, nil
 	}
 
-	return filtered, err
+	return tasks, nil
 }
 
 // Find returns really all tasks that satisfy the query.
-func FindAll(query db.Q) ([]Task, error) {
+func FindAll(filter bson.M) ([]Task, error) {
```

Comment:
Sorry to unwind a lot of work but: we don't need to refactor _all_ of these, just the ones above where we want to add the DisplayOnly filter. For example, `FindAllOld(ByIds(oldIDs))` can be changed to `FindAllOld(db.Query(ByIds(oldIds)))` instead of making FindAllOld a function that expects a bson.M.

Reply:
i may cry if i have to refactor again 😭

Reply:
this one was so fast though! 🚀